BitwardenShared/UI/Vault/VaultItem/CipherItemState.swift

```diff
@@ -140,6 +143,26 @@ struct CipherItemState: Equatable {
         }
     }
 
+    /// Whether or not this item can be deleted by the user.
+    /// New permission model from PM-18091
+    var canBeDeletedPermission: Bool {
+        // backwards compatibility for old server versions
+        guard let cipherPermissions = cipher.permissions else {
+            return canBeDeleted
+        }
+        return cipherPermissions.delete
+    }
```

Comment:
🎨 I believe you don't need the extra property and you can also save making the checks in the views if you combine this property with `canBeDeleted`, i.e.: 
```swift
    /// Whether or not this item can be deleted by the user.
    var canBeDeleted: Bool {
        // New permission model from PM-18091
        if restrictItemDeletionFlagEnabled, let cipherPermissions = cipher.permissions {
            return cipherPermissions.delete
        }

        guard !collectionIds.isEmpty else { return true }
        return collections.contains { collection in
            guard let id = collection.id else { return false }
            return collection.manage && collectionIds.contains(id)
        }
    }
```